### ISSUE TITLE
Closing the Asset Editor after opening a file with an improper name may crash the Editor

### ISSUE THREAD

**Issue by LB-KacperBieniek** [o3de/o3de#3409]
**Describe the bug**
It is possible to save all Asset Editor files (such as File Tag, Input Bindings, Script Events) with an improper name, for example _©.inputbindings_. The files do not appear in the _Open_ menu, but are available in the _Open Recent_ menu and trying to open them this way results in errors thrown in the Console:
```
[Error] (AssetDatabase) - No handler was registered for this asset (id={DAF0A851-3779-5B01-A2B3-E56CA2F664E2}:0, type={00000000-0000-0000-0000-000000000000})!
[Error] (AssetManager) - BlockUntilLoadComplete must be called after an asset has been queued for load.  Asset  ({00000000-0000-0000-0000-000000000000}:0) is not queued for load
```
After closing and opening the Asset Editor again, and trying to open the files with an improper name thru _Open Recent_ menu,  the same error messages are thrown, but this time, closing the Asset Editor prompts the user to save the file despite the fact, that it does not appear as open. Saving the file results in Editor crash.

Please refer to the video and attached files for more details.

**To Reproduce**
1. Open the Asset Editor (Tools > Asset Editor).
2. Create an Input Bindings file (File > New > Input Bindings).
3. Save the file with an improper name, for example _©.inputbindings_.
4. Close and reopen the Asset Editor.
5. Open the file saved in step 3 via _Open Recent_ menu (File > Open Recent). Notice the errors in the Console.
6. Try to close the Asset Editor.
7. On the "Save Changes?" prompt choose "Yes".
8. The Editor crashes.

**Expected behavior**
The Editor is stable. User is unable to save files with an improper name.

**Video**

https://user-images.githubusercontent.com/87059746/130604950-2617974b-1fe9-40d7-9e6f-6c4ef895cf72.mp4

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU Nvidia RTX 2060 Super
 - Memory 16GB

**Attachments:**
[error.log](https://github.com/o3de/o3de/files/7038612/error.log)
[error.zip](https://github.com/o3de/o3de/files/7038613/error.zip)
[Editor.log](https://github.com/o3de/o3de/files/7038614/Editor.log)

**Comment by alexmontAmazon** [comment]
@cgalvan could you take a look at this please? If you think there's a better fit for fixing this, feel free to reassign.

**Comment by LB-KacperBieniek** [comment]
Issue verified as fixed on Development (102c321) build.
